pkg/envtest/server.go

```diff
@@ -221,6 +227,10 @@ func (te *Environment) Start() (*rest.Config, error) {
 			if err != nil {
 				return nil, fmt.Errorf("unable to get configuration for existing cluster: %w", err)
 			}
+			te.KubeConfig, err = controlplane.KubeConfigFromREST(te.Config)
+			if err != nil {
+				return nil, fmt.Errorf("unable to convert configuration to kubeconfig: %w", err)
+			}
 		}
 	} else {
 		apiServer := te.ControlPlane.GetAPIServer()
```

Comment:
Let's add some test coverage. I would suggest envtest_test.go. Maybe simply at the end of BeforeEach (~l.55).

Let's take env.KubeConfig convert it via `clientcmd.RESTConfigFromKubeConfig` create an additional client from it and run a simple request (e.g. list CRD) to ensure the client works

Reply:
Makes sense.. this was old PR and no longer priority but may circle back . Anyone feel free to take over / push to this branch etc.

Reply:
No problem. I'll do that